scaleway/helpers.go

```diff
@@ -464,3 +466,42 @@ var UUIDRegex = regexp.MustCompile(`[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{
 func isUUID(s string) bool {
 	return UUIDRegex.MatchString(s)
 }
+
+// newTemplateFunc take a go template string a return a function that can be called to execute template.
+func newTemplateFunc(tplStr string) func(data interface{}) string {
+	t := template.Must(template.New("tpl").Parse(tplStr))
+	return func(tplParams interface{}) string {
+		buffer := bytes.Buffer{}
+		err := t.Execute(&buffer, tplParams)
+		if err != nil {
+			panic(err)
+		}
+		return buffer.String()
+	}
+}
+
+// testAccGetResourceAttr can be used in accptence tests to extract value from state and store it in dest
+func testAccGetResourceAttr(resourceName string, attrName string, dest *string) resource.TestCheckFunc {
+	return func(state *terraform.State) error {
+		r, exist := state.RootModule().Resources[resourceName]
+		if !exist {
+			return fmt.Errorf("unknown ressource %s", resourceName)
+		}
+
+		a, exist := r.Primary.Attributes[attrName]
+		if !exist {
+			return fmt.Errorf("unknown ressource %s", resourceName)
+		}
+
+		*dest = a
+		return nil
+	}
+}
+
+// testCheckResourceAttr is similar so standard resource.testCheckResourceAttr except value is a pointer and value can change
+// Can be useful when combined with testAccGetResourceAttr
+func testCheckResourceAttr(resourceName string, attrName string, value *string) resource.TestCheckFunc {
```

Comment:
You should use this https://godoc.org/github.com/hashicorp/terraform/helper/resource#TestCheckResourceAttrPtr